CorsixTH/Lua/research_department.lua

```diff
@@ -543,6 +543,7 @@ function ResearchDepartment:discoverObject(object, automatic)
           end
         end
       end
+      UIBuildRoom:updateList()
```

Comment:
Think you want to get the instance if its open still, like you had before.
I assume this is why you have
```lua
  local app = TheApp.ui.app
```
in update list instead of self.ui. The code I had avoided the dependency but with much more overhead, another reason I didn't like it. But the simpler behaviour avoids it.